pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -642,7 +642,11 @@ type ArchiveStrategy struct {
 }
 
 // TarStrategy will tar and gzip the file or directory when saving
-type TarStrategy struct{}
+type TarStrategy struct {
+	// Specifies the gzip compression level to use for the artifact.
```

Comment:
Minor, but it's convention (at least in Argo) that docsrings start with the field name:
```suggestion
	// CompressionLevel specifies the gzip compression level to use for the artifact.
```